src/javascript/typescript-config.ts

```diff
@@ -468,6 +576,14 @@ export class TypescriptConfig extends Component {
     this.file = new JsonFile(project, fileName, {
       allowComments: true,
       obj: {
+        extends: () =>
+          // use string value for singular extension for TS<5.0 support;
+          // omit if no extensions.
+          this.extends.length === 1
+            ? this.extends[0]
+            : this.extends.length
+            ? this.extends
+            : undefined,
```

Comment:
One last request: Turn this into a private helper method `renderExtends()`
```suggestion
        extends: () => this.renderExtends(),
```

Reply:
:+1: I wasn't the biggest fan of the nested ternary anyways :smile:

Reply:
Yes it's not... great. 😅